mars/dataframe/arithmetic/utils.py

```diff
@@ -46,7 +43,11 @@ def infer_index_value(left_index_value, right_index_value, operator):
         return left_index_value
 
     left_index = left_index_value.to_pandas()
+    if isinstance(left_index, pd.RangeIndex):
+        left_index = pd.RangeIndex(0)
     right_index = right_index_value.to_pandas()
+    if isinstance(right_index, pd.RangeIndex):
+        right_index = pd.RangeIndex(0)
     out_index = operator(left_index, right_index)
```

Comment:
I still think it makes no sense to do `out_index = operator(left_index, right_index)`, for example, when one of the index is `DatatimeIndex` and the other is `Float64Index`, the `operator` may cannot be applied.

I recommend to do something like `parse_index(left_index.join(right_index, how='outer'))` to get the correct type of result index, then we don't need to concern any exceptions even when the two indexes are both `RangeIndex` and even they have the same `slice` value.

Reply:
Thanks for the suggestions, it is indeed a better way to infer index value.

Reply:
Still have some problems:
 ``` Python
In: pd.Int64Index([]).join(pd.Float64Index([]), how='outer')                                                         
Out: Int64Index([], dtype='int64')
```
expected dtype is `float64`, maybe we can use `find_common_type` in `pandas` to infer dtype.